### PR TITLE
new storageclass for premium tier integration test

### PR DESCRIPTION
**Pull request by leiyiz**
**What type of PR is this?**

**What this PR does / why we need it**:
we need a new storage class for premium tier prow jobs

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: test/k8s-integration/config/fs-sc-premium-default-netwrok.yaml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+apiVersion: storage.k8s.io/v1
```
Comment (mattcary) [inline test/k8s-integration/config/fs-sc-premium-default-netwrok.yaml:1]: Typo in the filename (netwrok)

File: test/k8s-integration/config/fs-sc-premium-default-netwrok.yaml [hunk 2]
```diff
@@ -0,0 +1,8 @@
...
+kind: StorageClass
+metadata:
+  name: csi-filestore-premium
+provisioner: filestore.csi.storage.gke.io
+parameters:
+  tier: premium
+volumeBindingMode: WaitForFirstConsumer
```
Comment (mattcary) [inline test/k8s-integration/config/fs-sc-premium-default-netwrok.yaml:8]: This should probably be immediate binding mode? That will speed up the test slightly. Plus we'll probably want to move to immediate binding mode for the default deployment anyway.

**Review by mattcary — COMMENTED** [review]
Nit: some extra words in the PR title, 'new storageclass for premium tier integration test' would probably be better.

**Comment by mattcary** [comment]
> Nit: some extra words in the PR title, 'new storageclass for premium tier integration test' would probably be better.

(and in the commit message)

File: test/k8s-integration/config/fs-sc-premium-default-netwrok.yaml (hunk 1 above)
Comment (leiyiz) [reply test/k8s-integration/config/fs-sc-premium-default-netwrok.yaml]: Done

File: test/k8s-integration/config/fs-sc-premium-default-netwrok.yaml (hunk 2 above)
Comment (leiyiz) [reply test/k8s-integration/config/fs-sc-premium-default-netwrok.yaml]: Done

**Pull request merged**